bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/client/BinRpcClient.java

```diff
@@ -51,7 +51,9 @@ protected RpcRequest<byte[]> createRpcRequest(String methodName) {
 
     @Override
     protected String getRpcCallbackUrl() {
-        return "binary://" + config.getCallbackHost() + ":" + config.getBinCallbackPort();
+        String callbackUrl = "binary://" + config.getCallbackHost() + ":" + config.getXmlCallbackPort();
+        callbackUrl = callbackUrl.replaceAll("\s", "");
```

Comment:
Usually a space is url encoded to `%20` are you sure you want to strip them instead of encode?

Did you also consider to strip all whitespaces with `replaceAll("\\s+","")` ?

Reply:
Also original code calls `getBinCallbackPort` not `getXmlCallbackPort`

Reply:
Instead of correcting the configuration, I suggest to validate the configuration in the handler `initialize` method and set the thing state to CONFIGURATION_ERROR if it contains white spaces.